### ISSUE TITLE
Kernel 5.3.0-23-generic on Ubuntu hangs/crashes with v5.6.4.2

### ISSUE THREAD

**Issue by pdelagrave** [aircrack-ng/rtl8812au#486]
Hi,

Tried the latest code 3b0b7b989dd90e3debaa576ddbbbb0dedfce248b on Ubuntu with kernel v5.3.0 and the module loads fine but everything start hanging afterwards, even making it impossible to do a clean reboot.

* No secure boot issue.
* No compilation issue.
* Device is a Alfa AC1200 - AWUS036ACM.
* This is the only adapter on the machine.
* This is not a virtual machine.
* Tried both `make, insmod` and the `dkms` method.
* Using Ubuntu 19.10 if that changes anything.
* Had the same problem on a Linux Mint machine using a 4.15.0 kernel.

Here is the relevant part from `dmesg`:
```
13:24:43 desktop kernel: [  340.003483] usbcore: registered new interface driver 88XXau
13:24:49 desktop kernel: [  346.741328] usb 1-6: new high-speed USB device number 15 using xhci_hcd
13:24:49 desktop kernel: [  346.889810] usb 1-6: New USB device found, idVendor=0bda, idProduct=8812, bcdDevice= 0.00
13:24:49 desktop kernel: [  346.889815] usb 1-6: New USB device strings: Mfr=1, Product=2, SerialNumber=3
13:24:49 desktop kernel: [  346.889819] usb 1-6: Product: 802.11n NIC
13:24:49 desktop kernel: [  346.889821] usb 1-6: Manufacturer: Realtek
13:24:49 desktop kernel: [  346.889824] usb 1-6: SerialNumber: 123456
13:24:50 desktop kernel: [  347.012327] usb 1-6: 88XXau 00:a0:aa:aa:ba:ab hw_info[d7]
13:24:50 desktop systemd[1]: Starting Load/Save RF Kill Switch Status...
13:24:50 desktop iwd[1069]: Wiphy: 0, Name: phy0
13:24:50 desktop iwd[1069]: #011Permanent Address: 00:00:00:00:00:00
13:24:50 desktop iwd[1069]: #011Bands: 2.4 GHz 5 GHz
13:24:50 desktop iwd[1069]: #011Ciphers: CCMP TKIP BIP
13:24:50 desktop iwd[1069]: #011Supported iftypes: ad-hoc station ap
13:24:50 desktop mtp-probe: checking bus 1, device 15: "/sys/devices/pci0000:00/0000:00:14.0/usb1/1-6"
13:24:50 desktop mtp-probe: bus: 1, device: 15 was not an MTP device
```

Any suggestion to how and where to collect more logs would be appreciated.
Thanks

**Comment by terencode** [comment]
Same with an archer T4U on ArchLinux and kernel 5.3.8, I remember seeing something along ` "worker wlan0" is taking a long time`

**Comment by eugeneware** [comment]
I can confirm the same issue with Kal Linux running 5.3.0:

```
uname -a
Linux kali 5.3.0-kali2-amd64 #1 SMP Debian 5.3.9-3kali1 (2019-11-20) x86_64 GNU/Linux
```
I'm seeing it with multiple wifi cards running the rtl8812au chipset.

I can connect to wifi, but when I try to put it into monitor mode the machine halts.

**Comment by gpolitis** [comment]
I can also confirm the same issue with Ubuntu 18.04.4 LTS:

> uname -a
> Linux ubuntu-desktop 5.3.0-28-generic #30~18.04.1-Ubuntu SMP Fri Jan 17 06:14:09 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

I have a couple of TP-Link AC600 devices, one connecting to my WiFi and another acting as a Hotspot.

> lsusb
> Bus 001 Device 003: ID 2357:011f
> Bus 001 Device 002: ID 2357:011f

The hangs started happening randomly when I configured my second adapter to act as a Hotspot.

**Comment by ikegami-t** [comment]
The PC hang up issue is caused on Ubuntu 20.04 Linux kernel 5.4.0-29-generic with the 8812au driver when connected to SSID by Buffalo - WI-U3-866D.
Also the issue was not happened on Ubuntu 20.04 Linux kernel 5.7.0-rc3+ built locally.
So it seems that the issue is fixed by the latest kernel driver.
Now I am confirm again with the latest kernel.

**Comment by ikegami-t** [comment]
Just confirmed again as the 8812au driver works correctly with the kernel 5.7.0-rc5+ without any error.

**Comment by ikegami-t** [comment]
Not caused on the kernel 5.6.1 built locally.

**Comment by ikegami-t** [comment]
Caused on the kernel 5.5.1 built locally so seems that fixed by the kernel 5.6.1.

**Comment by ikegami-t** [comment]
Hi,
Just confirmed that the issue has been fixed by the patch in 5.6.1 below.
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/net/wireless?h=v5.6.1&id=9951ebfcdf2b97dbb28a5d930458424341e61aa2
Also it is applied to the v5.5.8 maintenance version.

**Comment by cedricbambooza** [comment]
pls consider closing the issue, when it's solved by now :)

**Comment by cyrus104** [comment]
Please stop spamming this project and politely message or email the maintainers.

**Comment by cedricbambooza** [comment]
I'm sorry that my effort is considered as spamming. 
May I ask where I can find a list of the Maintainers to this project?
The only one I found by now is kimocoder and he is maintaining somewhat 600 projects. 
when I look at how old some open issues are I think he has better things to do than closing old issues. 
So I don't think I'm spamming anything by politely asking the issue openers to consider closing there stuff.
I, at the moment, have the time to remind them and with more issues closed one can focuse on the important still opened ones.

**Comment by conqueror8888** [comment]
Yo bro, u were said to be spamming for a good reason!

Your email client burst 50+ identical messages into this list. You should have it checked out!

No one here is giving you a hard time at all!


Peace! ✌



-------- Original message --------
From: cedricbambooza ***@***.***>
Date: 5/11/21 05:02 (GMT-08:00)
To: aircrack-ng/rtl8812au ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [aircrack-ng/rtl8812au] Kernel 5.3.0-23-generic on Ubuntu hangs/crashes with v5.6.4.2 (#486)


I'm sorry that my effort is considered as spamming.
May I ask where I can find a list of the Maintainers to this project?
The only one I found by now is kimocoder and he is maintaining somewhat 600 projects.
when I look at how old some open issues are I think he has better things to do than closing old issues.
So I don't think I'm spamming anything by politely asking the issue openers to consider closing there stuff.
I, at the moment, have the time to remind them and with more issues closed one can focuse on the important still opened ones.

**Comment by cedricbambooza** [comment]
Yes, check out the content of the message. 
It's easier to use the same simple message over and over again rather than writing one for every issue whlie the content still is the same. 
If you get a notification for every comment in every issue I made I'm sorry. 
I thought only participants of that issue were getting a notification. 
But I'm not worrying about getting a hard time, just calm down. 
If people would close their issues in time, no one would have to be the garbage man.
Consider that as well.

**Comment by cyrus104** [comment]
Anyone that monitors this project was getting those email. I since have unsubscribed and will check on Kimocoder's twitter for big updates.

The maintainers are able to bulk close issues and say OBE if they want to. You haven't used github or any community style platform if you expect people to close their own tickets. Sadly that's a part of how github is designed to work, allow people to submit an issues and expect people to abandon it when they get distracted.

As a maintainer, being the "garbage man" is easy to cleanup. It's much harder to read through them all and try to test each case. If you have time on your hands like you say, try to test some of the issues on your own and report back, then suggest the issue be closed.

**Comment by kimocoder** [comment]
Zzzzzzzzzzzzzzz

**Comment by CGarces** [comment]
Hi @cedricbambooza 

I appreciate your efforts bu this kind of massive messages are annoying.
If you want to contribute, please open a new issue requesting  triage permissions.

https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-permission-levels-for-an-organization

That should allow  review and close the issues without write permissions to the repo.

**Comment by cedricbambooza** [comment]
Yes alright I'm sorry that this happend like it happened. 
By now I also found out, that one can subscribe to the whole project and now I see why it was an annoying thing for me to do. 
I can't understand how anyone who subscrribed to the whole project can keep track of everything coming in, but that's another story. 
What I have to clear up is that I don't want to force close issues that are open for a long time or issues at all. 
It should've only been a kind reminder to review old cases and consider closing them if possible. 
It just escalated on a bigger scale. 
In most cases I wouldn't be of any help either, the topics are simply to specific. 
So reviewing and testing them in the time I have instead of just reminding like I did, maybe in a less penetrant manner in the future,  isn't something I consider for maybe more than 5% of the issues.
I also don't think I will be around here any longer so the issue requesting triage permissions is probably also nothing for me to consider. 
But still thx for the hint @CGarces ;)

Good luck with the project! ;)
